applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp

```diff
@@ -300,36 +279,40 @@ void NodalConcentratedElement::CalculateRightHandSide(VectorType& rRightHandSide
     const unsigned int dimension = GetGeometry().WorkingSpaceDimension();
 
     // Resizing as needed the RHS
-    const unsigned int SystemSize = dimension;
+    const unsigned int system_size = dimension;
 
-    if ( rRightHandSideVector.size() != SystemSize )
-    {
-        rRightHandSideVector.resize( SystemSize, false );
-    }
+    if ( rRightHandSideVector.size() != system_size )
+        rRightHandSideVector.resize( system_size, false );
 
-    rRightHandSideVector = ZeroVector( SystemSize ); //resetting RHS
+    rRightHandSideVector = ZeroVector( system_size ); //resetting RHS
 
-    const array_1d<double, 3 > Current_Displacement = GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT);
-    array_1d<double, 3 > Volume_Acceleration = ZeroVector(3);
+    const array_1d<double, 3 >& current_displacement = GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT);
+    array_1d<double, 3 > volume_acceleration = ZeroVector(3);
 
     if( GetGeometry()[0].SolutionStepsDataHas(VOLUME_ACCELERATION) )
-    {
-        Volume_Acceleration = GetGeometry()[0].FastGetSolutionStepValue(VOLUME_ACCELERATION);
-    }
+        volume_acceleration = GetGeometry()[0].FastGetSolutionStepValue(VOLUME_ACCELERATION);
 
     // Compute and add external forces
-    const double Nodal_Mass = Element::GetValue(NODAL_MASS);
-    for ( unsigned int j = 0; j < dimension; j++ )
-    {
-        rRightHandSideVector[j]  += Volume_Acceleration[j] * Nodal_Mass;
+    if (this->Has(NODAL_MASS)){
```

Comment:
you can do it nicer by greating a const reference to the element;
the important point is that if you call the "const" version of the method and the variable is not there a zero is given back without allocating something new in the database

in short what i propose doing is:
     
      const auto& rconst_this = *this;
      rconst_this.GetValue(NODAL_MASS);

Reply:
I didn't know this kind of magic, but it has been probably the revelation of the year :heart_eyes: